scater-calculate-qc-metrics.R

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env Rscript 
+
+# Load optparse we need to check inputs
+suppressPackageStartupMessages(require(optparse))
+
+# Load common functions
+suppressPackageStartupMessages(require(workflowscriptscommon))
+
+# parse options
+option_list = list(
+  make_option(
+    c("-i", "--input-object-file"),
+    action = "store",
+    default = NA,
+    type = 'character',
+    help = "File name in which a serialized R SingleCellExperiment object where object matrix found"
+  ),
+  make_option(
+    c("-e", "--exprs-values"),
+    action = "store",
+    default = 'counts',
+    type = 'character',
+    help= "character string indicating which slot of the assayData from the 'SingleCellExperiment' object should be used to compute log-transformed expression values. Valid options are 'counts', 'tpm', 'cpm' and 'fpkm'. Defaults to the first available value of the options in the order shown."
+  ),
+  make_option(
+    c("-f", "--feature-controls"),
+    action = "store",
+    default = NULL,
+    type = 'character',
+    help = "comma-separated list of files with one feature per line. A named list is created containing one or more vectors to identify feature controls (for example, ERCC spike-in genes, mitochondrial genes, etc)"
```

Comment:
We should probably have discussed this, but can we explicitly require these to be character vectors (adjusting help text accordingly)? Less ambiguous than allowing logical or numeric indices.

Reply:
So, if you're fixing this at one file, you need to change the description here too!

Reply:
Hi @suhaibMo , the help text for this still says comma-separated list of files, while your implementation is a single file containing a list of the control files.